File: src/archipelago/impl/test/rr0/im/business/report/RoswellReportTest.java
package org.rr0.im.business.report;

import junit.framework.TestCase;
import org.rr0.database.model.report.Case;
import org.rr0.database.model.report.CaseImpl;
import org.rr0.database.model.report.Source;
import org.rr0.database.model.event.circumstance.*;
import org.rr0.database.model.event.SightingImpl;
import org.rr0.database.model.event.Event;
import org.rr0.database.model.actor.*;
import org.rr0.database.model.evaluation.classification.Category;
import org.rr0.database.model.evaluation.AbstractFunction;

/**
 * A test for describing the Roswell case using the RR0 Information Model.
 *
 * @author Jérôme Beau
 * @version 21 avr. 2003 16:18:28
 */
public class RoswellReportTest extends TestCase {

    public RoswellReportTest(String someTestName) {
        super(someTestName);
    }

    protected void setUp() throws InstantiationException, IllegalAccessException, ClassNotFoundException {

        // The case in itself
        Case roswellCase = new CaseImpl("The Roswell Incident");

        // Case's actors
        Identity brazelIdentity = new IdentityImpl("William W. Brazel");
        HumanBeing brazel = new HumanBeingImpl(brazelIdentity);
        Identity usafIdentity = new IdentityImpl("United States Air Force");
        Organization usaf = new OrganizationImpl(usafIdentity);

        // First case's event : deflagration
        Moment firstDeflagrationMoment = new PreciseMomentImpl(1947, 7, 2, 21, 50);
        EarthLocation roswellCityLocation = new EarthLocationImpl(43, OrientationImpl.NORTH, 142, OrientationImpl.WEST);
        Place roswellCity = new PlaceImpl("Roswell", roswellCityLocation);
        Location ranchLocation = new DeltaLocationImpl(roswellCity.getLocation(), OrientationImpl.NORTH_WEST, LengthImpl.UNKNOWN);
        Place brazelRanch = new PlaceImpl("Brazel's ranch", ranchLocation);
        Clouds firstDeflagrationClouds = new CloudsImpl(0);
        Temperature firstDeflagrationTemperature = new TemperatureImpl(15);
        Wind firstDeflagrationWind = new WindImpl(10);
        Category windCategory = (Category) AbstractFunction.getFunction("ViolentWind");
        Precipitations firstDeflagrationPrecipitations = new PrecipitationsImpl(0);
        WeatherCondition weatherCondition = new WeatherConditionImpl(firstDeflagrationClouds, firstDeflagrationTemperature, firstDeflagrationWind, firstDeflagrationPrecipitations);

        Source source = null;
        Event firstDeflagation = new SightingImpl("Brazel heards a deflagration", firstDeflagrationMoment, brazelRanch, brazel, source);
        roswellCase.getAccounts().add(firstDeflagation);

        // Second case's event : Mac brazel
    }

    public void test1() {

    }

    protected void tearDown() {
        // Nettoyage fin de test
    }
}

File: src/database/rr0/model/report/RoswellReport.test.ts
import {IdentityImpl} from "archipelago/impl/model/IdentityImpl"
import {Identity} from "archipelago/api/model/Identity"
import {Source} from "database/rr0/model/report/Source"
import {Case} from "database/rr0/model/report/Case"
import {HumanBeing} from "database/rr0/model/actor/HumanBeing"
import {Organization} from "database/rr0/model/actor/Organization"
import {AbstractFunction} from "database/rr0/model/evaluation/AbstractFunction"
import {PreciseMoment} from "database/rr0/model/event/circumstance/PreciseMomentImpl"
import {EarthLocation} from "database/rr0/model/event/circumstance/EarthLocation"
import {Orientation} from "database/rr0/model/event/circumstance/Orientation"
import {Place} from "database/rr0/model/event/circumstance/Place"
import {DeltaLocation} from "database/rr0/model/event/circumstance/DeltaLocation"
import {UNKNOWN_LENGTH} from "database/rr0/util/Length"
import {Clouds} from "database/rr0/model/event/circumstance/Clouds"
import {Temperature} from "database/rr0/model/event/circumstance/Temperature"
import {CelciusDegrees} from "database/rr0/util/Unit"
import {Wind} from "database/rr0/model/event/circumstance/Wind"
import {Precipitations} from "database/rr0/model/event/circumstance/Precipitations"
import {WeatherCondition} from "database/rr0/model/event/circumstance/WeatherCondition"
import {Category} from "database/rr0/model/evaluation/classification/Category"
import {Sighting} from "database/rr0/model/event/Sighting"

/**
 * A test for describing the Roswell case using the RR0 Information Model.
 */
describe("Report", () => {

  // The case in itself
  const roswellCase = new Case("The Roswell Incident")

  // Case's actors
  const brazelIdentity: Identity = new IdentityImpl("William W. Brazel")
  const brazel = new HumanBeing(brazelIdentity)
  const usafIdentity: Identity = new IdentityImpl("United States Air Force")
  const usaf = new Organization(usafIdentity)
  console.log(usaf)

  // First case's event : deflagration
  const firstDeflagrationMoment = new PreciseMoment(1947, 7, 2, 21, 50)
  const roswellCityLocation = new EarthLocation(43, Orientation.NORTH, 142, Orientation.WEST)
  const roswellCity = new Place("Roswell", roswellCityLocation)
  const ranchLocation = new DeltaLocation(roswellCity.location, Orientation.NORTH_WEST, UNKNOWN_LENGTH)
  const brazelRanch: Place = new Place("Brazel's ranch", ranchLocation)
  const firstDeflagrationClouds = new Clouds(0)
  const firstDeflagrationTemperature = new Temperature(CelciusDegrees, 15)
  const firstDeflagrationWind = new Wind(10)
  const windCategory = AbstractFunction.getFunction("ViolentWind") as Category
  console.log(windCategory)
  const firstDeflagrationPrecipitations = new Precipitations(0)
  const weatherCondition = new WeatherCondition(firstDeflagrationClouds, firstDeflagrationTemperature,
    firstDeflagrationWind, firstDeflagrationPrecipitations)
  console.log(weatherCondition)

  const source: Source | undefined = undefined
  const firstDeflagration = new Sighting("Brazel heards a deflagration", firstDeflagrationMoment, brazelRanch,
    brazel, source)
  roswellCase.accounts.push(firstDeflagration)

  // Second case's event : Mac brazel
})
